lib/routes/account.js

```diff
@@ -891,12 +893,37 @@ module.exports = function (
       handler: function (request, reply) {
         log.begin('Account.RecoveryEmailStatus', request)
         var sessionToken = request.auth.credentials
-        reply(
-          {
-            email: sessionToken.email,
-            verified: sessionToken.emailVerified
+
+        cleanUpIfAccountInvalid()
+          .then(createResponse)
+          .done(reply, reply)
+
+        function createResponse() {
+          return {
+             email: sessionToken.email,
+             verified: sessionToken.emailVerified
+           }
+        }
+
+        function cleanUpIfAccountInvalid() {
+          // Some historical bugs mean we've allowed creation
+          // of accounts with invalid email addresses. These
+          // can never be verified, so the best we can do is
+          // to delete them so the browser will stop polling.
+          if (!sessionToken.emailVerified) {
```

Comment:
I notice this repo sometimes has a space between `!` and its operand and sometimes doesn't. Do we expressly prefer one or the other.